### PR TITLE
DB-10702 improve FK DB2 errors compatibility.

### PR DESCRIPTION
**Pull request by hatyo**
The holy grail of this PR is to assign `LANG_INVALID_FK_COL_FOR_SETNULL` severity [-629](https://www.ibm.com/support/knowledgecenter/SSEPEK_11.0.0/codes/src/tpc/n629.html). The rest is just cleanup and dead code removal.

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java [hunk 1]
```diff
@@ -567,6 +567,9 @@ else if (sqlState.equals(SQLState.LANG_CHECK_CONSTRAINT_VIOLATED)) {
         else if (sqlState.equals(SQLState.LANG_FK_VIOLATION)) {
             se.setSeverity(-530);
         }
+        else if (sqlState.equals(SQLState.LANG_INVALID_FK_COL_FOR_SETNULL)) {
```
Comment (ipraznik-splice) [inline db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java:570]: Is it possible to add IT for this error code? For SQLState.LANG_FK_VIOLATION there are at least some tests in LoadReplaceModeIT.

**Review by ipraznik-splice — CHANGES_REQUESTED** [review]

**Review by ascend1 — APPROVED** [review]

**Review by dgomezferro — APPROVED** [review]
I'm OK with the change once an IT gets added

**Review by jyuanca — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java (hunk 1 above)
Comment (carolp-503) [reply db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java:570]: I suggest that we Jira this as a separate item if it's going to take very long at all to construct the IT.

**Review by carolp-503 — APPROVED** [review]
+1.  

Check with @hatyo about adding a new IT per @ipraznik-splice 's comment.

File: db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java (hunk 1 above)
Comment (hatyo) [reply db-engine/src/main/java/com/splicemachine/db/impl/jdbc/TransactionResourceImpl.java:583]: Thanks, I added a test for `SQLState.LANG_FK_VIOLATION` in `ForeignKeyActionIT`.

**Review by ipraznik-splice — APPROVED** [review]

**Pull request merged**